libs/Microsoft.MixedReality.WebRTC.Native/src/interop/global_factory.cpp

```diff
@@ -131,14 +135,33 @@ void GlobalFactory::RemoveObject(ObjectType type, TrackedObject* obj) noexcept {
     if (it != alive_objects_.end()) {
       RTC_CHECK(it->second == type);
       alive_objects_.erase(it);
-      if (alive_objects_.empty()) {
-        ShutdownNoLock();
-      }
     }
   } catch (...) {
   }
 }
 
+bool GlobalFactory::TryShutdown() noexcept {
+  std::unique_ptr<GlobalFactory>& factory = GlobalFactory::MutableInstance();
+  if (!factory) {
+    return false;  // not shutdown by this call
+  }
+  try {
+    if (factory->TryShutdownImpl()) {
+      factory = nullptr;  // fixme : outside lock...
+      return true;
+    }
+  } catch (...) {

```

Comment:
Why not let the exception go to the caller?

Reply:
The internal C++ classes are only helpers to implement the C interop API, which cannot throw. So better catch it here than add try/catch block in 20 places in the C functions. Ideally we'd disable exceptions completely in this project.

Reply:
The one place in this review which calls TryShutdown is ignoring the return value. That looks suspicious.

Reply:
Ah you meant that one method in particular, not in general why we have try/catch blocks? Actually `RemoveRef()` is a member method so should call `TryShutdownImpl()` directly. I will remove `TryShutdown()`.